src/CMakeLists.txt

```diff
@@ -34,62 +34,28 @@ set(EXTERNAL_PROJECTS_BUILD_DIR ${CMAKE_BINARY_DIR}/extern) # default build dire
-include(../cmake/SetupRawHashLikeTarget.cmake)
-include(../cmake/SetupRUClient.cmake)
-include(../cmake/SetupPOD5.cmake)
-include(../cmake/SetupHDF5.cmake)
-include(../cmake/SetupSLOW5.cmake)
-include(../cmake/SetupTFLite.cmake)
+include(${CMAKE_SOURCE_DIR}/cmake/SetupCCacheMold.cmake)
+include(${CMAKE_SOURCE_DIR}/cmake/SetupRawHashLikeTarget.cmake)
 
 set(CMAKE_VERBOSE_MAKEFILE TRUE CACHE BOOL "verbose makefile" FORCE) # print compilation commands
 
 enable_ccache()
 set_alternate_linker(mold)
 
-# todo: currently can only compile rawhash2 or the below, not both
-# this happens because functions like setup_hdf5 define the hdf5 target and the relationship to the target name
-# rather than doing it separately
-set(build_cli ON)
+add_executable(rawhash2_builtin main.cpp)
+setup_rawhashlike_target(rawhash2_builtin)
 
-if (build_cli)
-    message(STATUS "Building CLI")
-    
-    set(TARGET_NAME rawhash2)
-    add_executable(${TARGET_NAME} main.cpp)
-    setup_rawhashlike_target(${TARGET_NAME})
-    setup_pod5(${TARGET_NAME})
-    setup_ruclient(${TARGET_NAME})
-    setup_hdf5(${TARGET_NAME})
-    setup_slow5(${TARGET_NAME})
-    setup_tflite(${TARGET_NAME})
+add_library(rawhash2_lib SHARED rawhash_wrapper.cpp rawhash_wrapper.hpp)
+setup_rawhashlike_target(rawhash2_lib)
 
-else()
-    message(STATUS "Building shared library")
+file(GLOB_RECURSE HEADER_FILES ${CMAKE_SOURCE_DIR}/src/*.h ${CMAKE_SOURCE_DIR}/src/*.hpp)
+set_property(TARGET rawhash2_lib PROPERTY PUBLIC_HEADER ${HEADER_FILES})
+set_target_properties(rawhash2_lib PROPERTIES INSTALL_RPATH $ORIGIN OUTPUT_NAME rawhash2)
+install(TARGETS rawhash2_lib
+        LIBRARY DESTINATION lib
+        PUBLIC_HEADER DESTINATION include/${PROJECT_NAME})
 
-    # set(TARGET_NAME rawhash2_wrapper_example)
-    # add_executable(${TARGET_NAME} wrapper_example.cpp rawhash_wrapper.cpp rawhash_wrapper.hpp)
-    set(TARGET_NAME rawhash2_wrapper)
-    add_library(${TARGET_NAME} SHARED rawhash_wrapper.cpp rawhash_wrapper.hpp)
-    setup_rawhashlike_target(${TARGET_NAME})
-    setup_pod5(${TARGET_NAME})
-    setup_ruclient(${TARGET_NAME})
-    setup_hdf5(${TARGET_NAME})
-    setup_slow5(${TARGET_NAME})
-    setup_tflite(${TARGET_NAME})
-
-    set(TARGET_NAME rawhash2_usinglib)
-    add_executable(${TARGET_NAME} wrapper_example.cpp)
-    target_link_libraries(${TARGET_NAME} PRIVATE rawhash2_wrapper)
-    # setup_rawhashlike_target(${TARGET_NAME})
-    # # setup_pod5(${TARGET_NAME})
-    # target_compile_definitions(${TARGET_NAME} PRIVATE NPOD5RH=1) # todo: workaround
-    # setup_ruclient(${TARGET_NAME})
-    # setup_hdf5(${TARGET_NAME})
-    # setup_slow5(${TARGET_NAME})
-    # setup_tflite(${TARGET_NAME})
-endif()
-
-
-# setup_pod5()
-# add_pod5_to_target(${TARGET_NAME})
+add_executable(rawhash2 wrapper_example.cpp)
+target_link_libraries(rawhash2 PRIVATE rawhash2_lib)
+set_target_properties(rawhash2 PROPERTIES INSTALL_RPATH $ORIGIN/../lib)
+install(TARGETS rawhash2 RUNTIME DESTINATION bin)
```

Comment:
Currently `make install` would install the shared library and also an executable that links to it. Should we install an executable that doesn't link to the rawhash2 shared library as well? Also generally what's the reason to build both targets?

Reply:
Yes, you can build a shared and static library. The python wrapper needs a shared library, if I remember well.
The important thing is that both the rawhash python wrapper and rawhash cli can be built at the same time, unlike previously.

Reply:
Yes, we should be able to build both currently. What I'm really asking is: During the install, should we install both the "standalone" client and the client that depends on the static library, or only one of them? If only one, which? Would it be okay to only install the client that depends on RawHash shared library?